crypto/encode_decode/decoder_lib.c

```diff
@@ -504,8 +525,9 @@ int OSSL_DECODER_CTX_add_extra(OSSL_DECODER_CTX *ctx,
                     = OSSL_DECODER_INSTANCE_get_input_type(decoder_inst);
 
 
-                OSSL_DECODER_do_all_provided(libctx,
-                                             collect_extra_decoder, &data);
+                for (j = 0; j < numdecoders; j++)
+                    collect_extra_decoder(sk_OSSL_DECODER_value(skdecoders, j),
```

Comment:
collect_extra_decoder looks like a bottle neck too -- it loops to check that it hasn't seen the decoder before & it does this individually for every addition.